app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,33 @@
 class PostsController < ApplicationController
   def show
     @post = Post.where(author_id: params[:user_id], id: params[:id]).first
+    @comments = @post.comments
   end
 
   def index
+    @user = User.find(params[:user_id])
     @posts = Post.where(author_id: params[:user_id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    @post.author = current_user
+    if @post.save
+      redirect_to user_posts_path(current_user)
+    else
+      render :new
+    end
+  end
+
```

Comment:
- [ ] When I tried to create a post from the form nothing happened the page gets reload and there is no post please make sure to fix this issue so that it will create the post without prob